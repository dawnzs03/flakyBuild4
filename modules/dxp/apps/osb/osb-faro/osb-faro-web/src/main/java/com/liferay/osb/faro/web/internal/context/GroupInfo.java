/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.context;

/**
 * @author Marcellus Tavares
 */
public class GroupInfo {

	public GroupInfo(long groupId) {
		_groupId = groupId;
	}

	public long getGroupId() {
		return _groupId;
	}

	private final long _groupId;

}