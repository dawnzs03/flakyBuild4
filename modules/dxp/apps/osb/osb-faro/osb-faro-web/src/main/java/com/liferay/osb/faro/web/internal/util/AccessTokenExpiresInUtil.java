/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

/**
 * @author Ivica Cardic
 */
public class AccessTokenExpiresInUtil {

	public static long getExpiresIn() {
		return _expiresIn;
	}

	public static void removeExpiresIn() {
		_expiresIn = 0;
	}

	public static void setExpiresIn(long expiresIn) {
		_expiresIn = expiresIn;
	}

	private static long _expiresIn;

}