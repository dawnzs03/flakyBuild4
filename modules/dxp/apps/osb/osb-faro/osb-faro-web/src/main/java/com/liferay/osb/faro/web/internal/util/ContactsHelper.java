/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.constants.FieldMappingConstants;
import com.liferay.osb.faro.engine.client.model.Individual;
import com.liferay.osb.faro.engine.client.model.IndividualSegment;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.osb.faro.web.internal.constants.FaroConstants;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualDisplay;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualSegmentDisplay;
import com.liferay.osb.faro.web.internal.model.display.main.FaroEntityDisplay;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Shinn Lok
 */
@Component(service = ContactsHelper.class)
public class ContactsHelper {

	public FaroEntityDisplay getContactsEntityDisplay(
		FaroProject faroProject, String contactsEntityId, int type) {

		if (type == FaroConstants.TYPE_INDIVIDUAL) {
			Individual individual = _contactsEngineClient.getIndividual(
				faroProject, contactsEntityId, null);

			return new IndividualDisplay(individual);
		}

		IndividualSegment individualSegment =
			_contactsEngineClient.getIndividualSegment(
				faroProject, contactsEntityId, false);

		return new IndividualSegmentDisplay(individualSegment);
	}

	public String getOwnerType(int type) {
		if (type == FaroConstants.TYPE_ACCOUNT) {
			return FieldMappingConstants.OWNER_TYPE_ACCOUNT;
		}
		else if (type == FaroConstants.TYPE_INDIVIDUAL) {
			return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL;
		}

		return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL_SEGMENT;
	}

	@Reference
	private ContactsEngineClient _contactsEngineClient;

}