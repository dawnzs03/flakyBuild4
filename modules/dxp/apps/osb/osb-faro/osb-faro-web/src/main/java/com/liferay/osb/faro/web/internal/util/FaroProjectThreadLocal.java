/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.model.FaroProject;
import com.liferay.petra.lang.CentralizedThreadLocal;

/**
 * @author Marcellus Tavares
 */
public class FaroProjectThreadLocal {

	public static FaroProject getFaroProject() {
		return _faroProject.get();
	}

	public static void removeFaroProject() {
		_faroProject.remove();
	}

	public static void setFaroProject(FaroProject faroProject) {
		_faroProject.set(faroProject);
	}

	private static final ThreadLocal<FaroProject> _faroProject =
		new CentralizedThreadLocal<>(
			FaroProjectThreadLocal.class + "._faroProject");

}