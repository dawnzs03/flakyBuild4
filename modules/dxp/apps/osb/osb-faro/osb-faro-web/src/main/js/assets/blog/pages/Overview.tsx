import AudienceReportCard from 'shared/components/audience-report/AudienceReportBaseCard';
import BlogMetricCard from 'assets/blog/components/BlogMetricCard';
import DevicesCard from 'assets/blog/hocs/DevicesCard';
import LocationsCard from 'assets/blog/hocs/LocationsCard';
import React from 'react';
import TouchpointsListCard from 'assets/hocs/TouchpointsListCard';
import {MetricName} from 'shared/types/MetricName';
import {Name} from 'shared/components/audience-report/types';

const Overview = () => (
	<>
		<div className='row'>
			<div className='col-sm-12'>
				<BlogMetricCard
					label={Liferay.Language.get('visitors-behavior')}
				/>
			</div>
		</div>

		<div className='row'>
			<div className='col-sm-12'>
				<AudienceReportCard
					knownIndividualsTitle={Liferay.Language.get(
						'segmented-views'
					)}
					query={{
						metricName: MetricName.Views,
						name: Name.Blog
					}}
					uniqueVisitorsTitle={Liferay.Language.get('views')}
				/>
			</div>
		</div>

		<div className='row'>
			<div className='col-lg-6 col-md-12'>
				<LocationsCard
					label={Liferay.Language.get('views-by-location')}
					legacyDropdownRangeKey={false}
				/>
			</div>

			<div className='col-lg-6 col-md-12'>
				<DevicesCard
					label={Liferay.Language.get('views-by-technology')}
					legacyDropdownRangeKey={false}
				/>
			</div>
		</div>

		<div className='row'>
			<div className='col-sm-12'>
				<TouchpointsListCard
					assetType='BLOG'
					label={Liferay.Language.get('asset-appears-on')}
					legacyDropdownRangeKey={false}
				/>
			</div>
		</div>
	</>
);

export default Overview;
