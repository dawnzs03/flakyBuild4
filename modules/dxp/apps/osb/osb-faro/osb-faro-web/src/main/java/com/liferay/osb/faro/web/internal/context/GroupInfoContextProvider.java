/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.context;

import com.github.scribejava.core.exceptions.OAuthException;

import com.liferay.expando.kernel.model.ExpandoBridge;
import com.liferay.oauth2.provider.model.OAuth2Authorization;
import com.liferay.oauth2.provider.service.OAuth2AuthorizationLocalService;

import java.util.Date;

import javax.servlet.http.HttpServletRequest;

import javax.ws.rs.ext.Provider;

import org.apache.cxf.jaxrs.ext.ContextProvider;
import org.apache.cxf.message.Message;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marcellus Tavares
 */
@Component(service = GroupInfoContextProvider.class)
@Provider
public class GroupInfoContextProvider implements ContextProvider<GroupInfo> {

	@Override
	public GroupInfo createContext(Message message) {
		HttpServletRequest httpServletRequest =
			(HttpServletRequest)message.getContextualProperty("HTTP.REQUEST");

		String authorization = httpServletRequest.getHeader("Authorization");

		if (authorization == null) {
			throw new IllegalStateException(
				"Authorization Header is not available");
		}

		try {
			OAuth2Authorization oAuth2Authorization =
				_oAuth2AuthorizationLocalService.
					getOAuth2AuthorizationByAccessTokenContent(
						authorization.substring(7));

			Date currentDate = new Date(System.currentTimeMillis());

			if (currentDate.after(
					oAuth2Authorization.getAccessTokenExpirationDate())) {

				throw new OAuthException(
					"Your access token is invalid. Please check your token " +
						"and try again.");
			}

			ExpandoBridge expandoBridge =
				oAuth2Authorization.getExpandoBridge();

			return new GroupInfo(
				(long)expandoBridge.getAttribute("groupId", false));
		}
		catch (Exception exception) {
			throw new IllegalStateException(
				"Unable to fetch the OAuth2Authorization with access token " +
					authorization.substring(7),
				exception);
		}
	}

	@Reference
	private OAuth2AuthorizationLocalService _oAuth2AuthorizationLocalService;

}