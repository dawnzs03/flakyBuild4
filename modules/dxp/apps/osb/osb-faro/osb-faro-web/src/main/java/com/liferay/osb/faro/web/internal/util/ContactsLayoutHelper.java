/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.contacts.model.ContactsCardTemplate;
import com.liferay.osb.faro.contacts.service.ContactsCardTemplateLocalService;
import com.liferay.osb.faro.web.internal.card.template.ContactsCardTemplateManagerHelper;
import com.liferay.osb.faro.web.internal.model.display.contacts.card.template.ContactsLayoutTemplateSettingDisplay;
import com.liferay.petra.string.StringBundler;
import com.liferay.petra.string.StringPool;
import com.liferay.portal.kernel.model.UserConstants;

import java.util.ArrayList;
import java.util.List;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Matthew Kong
 */
@Component(service = ContactsLayoutHelper.class)
public class ContactsLayoutHelper {

	public String addContactsCardTemplateIds(long groupId, int type)
		throws Exception {

		List<List<ContactsLayoutTemplateSettingDisplay>>
			contactsLayoutTemplateSettingDisplaysList = new ArrayList<>();

		for (int[] contactsCardTemplateTypes :
				_contactsCardTemplateManagerUtil.
					getDefaultContactsCardTemplateTypes(type)) {

			List<ContactsLayoutTemplateSettingDisplay>
				contactsLayoutTemplateSettingDisplays = new ArrayList<>();

			for (int contactsCardTemplateType : contactsCardTemplateTypes) {
				ContactsCardTemplate contactsCardTemplate =
					_contactsCardTemplateLocalService.addContactsCardTemplate(
						groupId, UserConstants.USER_ID_DEFAULT,
						_contactsCardTemplateManagerUtil.getDefaultName(
							contactsCardTemplateType),
						_contactsCardTemplateManagerUtil.getDefaultSettings(
							contactsCardTemplateType),
						contactsCardTemplateType);

				contactsLayoutTemplateSettingDisplays.add(
					new ContactsLayoutTemplateSettingDisplay(
						contactsCardTemplate.getContactsCardTemplateId(), 0));
			}

			contactsLayoutTemplateSettingDisplaysList.add(
				contactsLayoutTemplateSettingDisplays);
		}

		return JSONUtil.writeValueAsString(
			contactsLayoutTemplateSettingDisplaysList);
	}

	public String addHeaderContactsCardTemplateIds(long groupId, int type) {
		int[] contactsCardTemplateTypes =
			_contactsCardTemplateManagerUtil.
				getDefaultHeaderContactsCardTemplateTypes(type);

		if (contactsCardTemplateTypes.length == 0) {
			return StringPool.BLANK;
		}

		StringBundler sb = new StringBundler(
			contactsCardTemplateTypes.length * 2);

		for (int contactsCardTemplateType : contactsCardTemplateTypes) {
			ContactsCardTemplate contactsCardTemplate =
				_contactsCardTemplateLocalService.addContactsCardTemplate(
					groupId, UserConstants.USER_ID_DEFAULT,
					_contactsCardTemplateManagerUtil.getDefaultName(
						contactsCardTemplateType),
					_contactsCardTemplateManagerUtil.getDefaultSettings(
						contactsCardTemplateType),
					contactsCardTemplateType);

			sb.append(contactsCardTemplate.getContactsCardTemplateId());

			sb.append(StringPool.COMMA);
		}

		sb.setIndex(sb.index() - 1);

		return sb.toString();
	}

	@Reference
	private ContactsCardTemplateLocalService _contactsCardTemplateLocalService;

	@Reference
	private ContactsCardTemplateManagerHelper _contactsCardTemplateManagerUtil;

}