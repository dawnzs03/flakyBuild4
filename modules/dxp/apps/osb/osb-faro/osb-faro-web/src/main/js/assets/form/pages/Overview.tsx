import AudienceReportCard from 'shared/components/audience-report/AudienceReportBaseCard';
import DevicesCard from 'assets/form/hocs/DevicesCard';
import FormAbandonmentCard from 'assets/form/hocs/FormAbandonmentCard';
import FormMetricCard from 'assets/form/components/FormMetricCard';
import LocationsCard from 'assets/form/hocs/LocationsCard';
import React from 'react';
import TouchpointsListCard from 'assets/hocs/TouchpointsListCard';
import {ENABLE_FORM_ABANDONMENT} from 'shared/util/constants';
import {MetricName} from 'shared/types/MetricName';
import {Name} from 'shared/components/audience-report/types';

const Overview = () => (
	<>
		<div className='row'>
			<div className='col-sm-12'>
				<FormMetricCard
					label={Liferay.Language.get('visitors-behavior')}
					legacyDropdownRangeKey={false}
				/>
			</div>
		</div>

		<div className='row'>
			<div className='col-sm-12'>
				<AudienceReportCard
					knownIndividualsTitle={Liferay.Language.get(
						'segmented-submissions'
					)}
					query={{
						metricName: MetricName.Submissions,
						name: Name.Form
					}}
					segmentsTitle={Liferay.Language.get('submitter-segments')}
					uniqueVisitorsTitle={Liferay.Language.get('submissions')}
				/>
			</div>
		</div>

		<div className='row'>
			<div className='col-lg-6 col-md-12'>
				<LocationsCard
					label={Liferay.Language.get('submissions-by-location')}
					legacyDropdownRangeKey={false}
					metricLabel={Liferay.Language.get('submissions')}
				/>
			</div>

			<div className='col-lg-6 col-md-12'>
				<DevicesCard
					label={Liferay.Language.get('submissions-by-technology')}
					legacyDropdownRangeKey={false}
					metricLabel={Liferay.Language.get('submissions')}
				/>
			</div>
		</div>

		{ENABLE_FORM_ABANDONMENT && (
			<div className='row'>
				<div className='col-sm-12'>
					<FormAbandonmentCard
						label={Liferay.Language.get('form-abandonment')}
						legacyDropdownRangeKey={false}
					/>
				</div>
			</div>
		)}

		<div className='row'>
			<div className='col-sm-12'>
				<TouchpointsListCard
					assetType='FORM'
					label={Liferay.Language.get('asset-appears-on')}
					legacyDropdownRangeKey={false}
				/>
			</div>
		</div>
	</>
);

export default Overview;
