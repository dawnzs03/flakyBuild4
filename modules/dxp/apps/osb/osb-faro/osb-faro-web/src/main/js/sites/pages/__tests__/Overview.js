import BasePage from 'shared/components/base-page';
import client from 'shared/apollo/client';
import Overview from '../Overview';
import React from 'react';
import {ApolloProvider} from '@apollo/react-components';
import {ChannelContext} from 'shared/context/channel';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {MemoryRouter, Route} from 'react-router-dom';
import {mockChannelContext} from 'test/mock-channel-context';
import {Routes} from 'shared/util/router';

jest.unmock('react-dom');

const MOCK_CONTEXT = {
	rangeKey: {defaultValue: '30'},
	router: {
		params: {
			channelId: '123123',
			groupId: '23'
		},
		query: {
			rangeKey: '30'
		}
	}
};

describe('Sites Dashboard Overview', () => {
	afterEach(cleanup);

	it('render', () => {
		const {container, getAllByText, getByText} = render(
			<ApolloProvider client={client}>
				<MemoryRouter initialEntries={['/workspace/23/123123/sites']}>
					<Route path={Routes.SITES}>
						<ChannelContext.Provider value={mockChannelContext()}>
							<BasePage.Context.Provider value={MOCK_CONTEXT}>
								<Overview
									channelName='Test Channel'
									router={{
										params: {
											channelId: '123123',
											groupId: '23'
										}
									}}
								/>
							</BasePage.Context.Provider>
						</ChannelContext.Provider>
					</Route>
				</MemoryRouter>
			</ApolloProvider>
		);
		fireEvent.click(getByText('All Visitors'));

		expect(getAllByText('All Visitors')[1]).toBeTruthy();
		expect(getByText('Anonymous Visitors')).toBeTruthy();
		expect(getByText('Known Visitors')).toBeTruthy();

		expect(container).toMatchSnapshot();
	});
});
