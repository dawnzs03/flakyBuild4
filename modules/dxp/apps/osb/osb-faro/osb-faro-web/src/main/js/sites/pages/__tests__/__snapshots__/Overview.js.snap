// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sites Dashboard Overview render 1`] = `
<div>
  <div
    class="sites-dashboard-overview-root overview-root"
  >
    <div
      aria-hidden="true"
      class="row"
      data-aria-hidden="true"
    >
      <div
        class="col-xl-12"
      >
        <div
          class="card card-root analytics-metrics-card"
          style="min-height: 605px;"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Test Channel Activities
            </h5>
            <div
              class="d-flex"
            >
              <div
                class="interval-selector-root mr-3 btn-group"
                role="group"
              >
                <button
                  class="button-root interval-option text-uppercase active btn btn-sm btn-secondary"
                  type="button"
                >
                  D
                </button>
                <button
                  class="button-root interval-option text-uppercase btn btn-sm btn-secondary"
                  type="button"
                >
                  W
                </button>
                <button
                  class="button-root interval-option text-uppercase btn btn-sm btn-secondary"
                  type="button"
                >
                  M
                </button>
              </div>
              <span
                aria-hidden="true"
                class="loading-root loading-animation"
              />
            </div>
          </div>
          <div
            class="card-body analytics-metrics"
          >
            <span
              aria-hidden="true"
              class="loading-root spacer loading-animation"
            />
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="row"
      data-aria-hidden="true"
    >
      <div
        class="col-xl-6"
      >
        <div
          class="card card-root top-pages-card-root table-tabs-root"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Top Pages
            </h5>
            <span
              aria-hidden="true"
              class="loading-root loading-animation"
            />
          </div>
          <div
            class="card-body w-100 d-flex flex-column flex-grow-1 no-padding"
          >
            <ul
              class="card-tabs-root"
            >
              <li
                class="card-tab active"
                data-testid="visitorsMetric"
              >
                <button
                  class="button-root btn btn-unstyled"
                  type="button"
                >
                  <span
                    class="title"
                  >
                    Visited Pages
                  </span>
                  <div />
                </button>
              </li>
              <li
                class="card-tab"
                data-testid="entrancesMetric"
              >
                <button
                  class="button-root btn btn-unstyled"
                  type="button"
                >
                  <span
                    class="title"
                  >
                    Entrance Pages
                  </span>
                  <div />
                </button>
              </li>
              <li
                class="card-tab"
                data-testid="exitRateMetric"
              >
                <button
                  class="button-root btn btn-unstyled"
                  type="button"
                >
                  <span
                    class="title"
                  >
                    Exit Pages
                  </span>
                  <div />
                </button>
              </li>
            </ul>
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
          <div
            class="card-footer"
          >
            <a
              class="button-root btn btn-outline-borderless btn-sm btn-outline-secondary"
              href="/workspace/23/123123/sites/pages?rangeKey=30"
            >
              View Pages
              <svg
                class="lexicon-icon lexicon-icon-angle-right-small icon-root ml-2"
                role="presentation"
              >
                <use
                  xlink:href="#angle-right-small"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-xl-6"
      >
        <div
          class="card card-root acquisitions-card-root table-tabs-root"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Acquisitions
            </h5>
            <span
              aria-hidden="true"
              class="loading-root loading-animation"
            />
          </div>
          <div
            class="card-body w-100 d-flex flex-column flex-grow-1 no-padding"
          >
            <ul
              class="card-tabs-root"
            >
              <li
                class="card-tab active"
                data-testid="CHANNEL"
              >
                <button
                  class="button-root btn btn-unstyled"
                  type="button"
                >
                  <span
                    class="title"
                  >
                    Channels
                  </span>
                  <div />
                </button>
              </li>
              <li
                class="card-tab"
                data-testid="SOURCE_MEDIUM"
              >
                <button
                  class="button-root btn btn-unstyled"
                  type="button"
                >
                  <span
                    class="title"
                  >
                    Source | Medium
                  </span>
                  <div />
                </button>
              </li>
              <li
                class="card-tab"
                data-testid="REFERRER"
              >
                <button
                  class="button-root btn btn-unstyled"
                  type="button"
                >
                  <span
                    class="title"
                  >
                    Referrers
                  </span>
                  <div />
                </button>
              </li>
            </ul>
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="row"
      data-aria-hidden="true"
    >
      <div
        class="col-xl-4"
      >
        <div
          class="card card-root visitors-by-time-card"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Visitors by Day and Time
            </h5>
            <span
              aria-hidden="true"
              class="loading-root loading-animation"
            />
          </div>
          <div
            class="card-body"
          >
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
        </div>
      </div>
      <div
        class="col-xl-4"
      >
        <div
          class="card card-root search-terms-card-root"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Search Terms
            </h5>
            <span
              aria-hidden="true"
              class="loading-root loading-animation"
            />
          </div>
          <div
            class="table-responsive table-root flex-grow-1"
          >
            <table
              class="table table-autofit table-hover table-heading-nowrap table-nowrap table-row-no-bordered"
            >
              <thead>
                <tr>
                  <th
                    class="table-head-title table-cell-expand"
                  >
                    Search Query | Searches
                  </th>
                  <th
                    class="table-head-title table-column-text-end"
                  >
                    % of Searches
                  </th>
                </tr>
              </thead>
            </table>
            <div
              class="loading-root overlay"
            >
              <span
                aria-hidden="true"
                class="loading-root center loading-animation"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xl-4"
      >
        <div
          class="card card-root interests-card-root"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Interests
            </h5>
            <span
              aria-hidden="true"
              class="loading-root loading-animation"
            />
          </div>
          <div
            class="table-responsive table-root flex-grow-1"
          >
            <table
              class="table table-autofit table-hover table-heading-nowrap table-nowrap table-row-no-bordered"
            >
              <thead>
                <tr>
                  <th
                    class="table-head-title table-cell-expand"
                  >
                    Interest Topics | Sessions
                  </th>
                  <th
                    class="table-head-title table-column-text-end"
                  >
                    % of Sessions
                  </th>
                </tr>
              </thead>
            </table>
            <div
              class="loading-root overlay"
            >
              <span
                aria-hidden="true"
                class="loading-root center loading-animation"
              />
            </div>
          </div>
          <div
            class="card-footer"
          >
            <a
              class="button-root btn btn-outline-borderless btn-sm btn-outline-secondary"
              href="/workspace/23/123123/sites/interests?rangeEnd=null&rangeKey=30&rangeStart=null"
            >
              All Interests
              <svg
                class="lexicon-icon lexicon-icon-angle-right-small icon-root ml-2"
                role="presentation"
              >
                <use
                  xlink:href="#angle-right-small"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="row"
      data-aria-hidden="true"
    >
      <div
        class="col-xl-6"
      >
        <div
          class="card card-root analytics-locations-card"
          style="min-height: 536px;"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Sessions by Location
            </h5>
            <div
              class="d-flex"
            >
              <span
                aria-hidden="true"
                class="loading-root loading-animation"
              />
            </div>
          </div>
          <div
            class="card-body"
          >
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
        </div>
      </div>
      <div
        class="col-xl-6"
      >
        <div
          class="card card-root analytics-devices-card"
          style="min-height: 536px;"
        >
          <div
            class="card-header align-items-center d-flex justify-content-between"
          >
            <h5
              class="card-title"
            >
              Session Technology
            </h5>
            <div
              class="d-flex"
            >
              <span
                aria-hidden="true"
                class="loading-root loading-animation"
              />
            </div>
          </div>
          <div
            class="card-body"
          >
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-xl-12"
      >
        <div
          class="card card-root cohort-analysis-card-root"
        >
          <div
            aria-hidden="true"
            class="card-header"
            data-aria-hidden="true"
          >
            <h5
              class="card-title"
            >
              Cohort Analysis
            </h5>
          </div>
          <div
            class="card-body"
          >
            <div
              class="form-group form-group-autofit"
            >
              <div
                class="form-group-item form-group-item-shrink"
              >
                <button
                  aria-activedescendant="visitors"
                  aria-expanded="true"
                  aria-haspopup="listbox"
                  aria-owns="clay-id-2"
                  class="form-control form-control-select form-control-select-secondary visitors-type-select show"
                  role="combobox"
                  style=""
                  tabindex="0"
                  type="button"
                >
                  All Visitors
                </button>
              </div>
              <div
                aria-hidden="true"
                class="form-group-item form-group-item-label form-group-item-shrink"
                data-aria-hidden="true"
              >
                Broken Down By
              </div>
              <div
                aria-hidden="true"
                class="form-group-item form-group-item-shrink"
                data-aria-hidden="true"
              >
                <button
                  aria-activedescendant=""
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="form-control form-control-select form-control-select-secondary interval-select"
                  role="combobox"
                  tabindex="0"
                  type="button"
                >
                  Day
                </button>
              </div>
            </div>
            <span
              aria-hidden="true"
              class="loading-root center loading-animation"
              data-aria-hidden="true"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
